chess-AI/train.py

```diff
@@ -15,73 +23,144 @@ class MctsTrain:
+        self.mcts_probs = []
+        self.mcts_evals = []
 
-    def training_episode(self, nnet):
-        # At each episode need to set the training example to empty
-        self.training_examples = []
+        # Stores board fen strings
+        self.boards = []
 
-        # The board needs to be initialized ot the starting state
-        board = chess.Board()
-        fen_string = board.fen()
+        self.mcts = Mcts(exploration)
+        self.policy_converter = PlayNetworkPolicyConverter()
 
-        while True:
-            # Perform mcts simulation
-            for _ in range(self.mcts_simulations):
-                self.mcts.search(board, nnet)
+        # Learning rate
+        self.lr = lr
 
-            # Need to get the moves and policy from the mcts
-            # NOTE: moves[i] corresponds to search_probs[i]
-            moves, search_probs = self.mcts.find_search_probs(fen_string)
+    def training_episode(self, nnet, epochs):
+        with torch.no_grad():
+            # At each episode need to set the training example and network predictions to empty
+            self.mcts_probs = []
+            self.mcts_evals = []
+            self.boards = []
 
-            # Gets a random index from the search_probs and makes random move
-            rand_move_idx = random.randint(0, len(search_probs) - 1)
-            move = moves[rand_move_idx]
+            batch_size = 8
+            num = 0
 
-            # Converts mcts search probabilites to (8,8,73) vector
-            full_search_probs = self.policy_converter.compute_full_search_probs(moves,
-                                                                                search_probs,
-                                                                                board)
-            
-            # Adds entry to the training examples
-            self.training_examples.append([fen_string, full_search_probs, None])
+            for _ in range(epochs):
+                # The board needs to be initialized to the starting state
+                board = chess.Board()
 
-            # Makes the random action on the board, and gets fen string
-            move = chess.Move.from_uci(move)
-            board.push(move)
-            fen_string = board.fen()
+                while True:
+                    fen_string = board.fen()
 
-            # if the game is over then that is the end of the episode
-            if board.is_game_over():
-                # Need to assign the rewards to the examples
-                self.assign_rewards(board)
-                return
+                    # Perform mcts simulations
+                    for _ in range(self.mcts_simulations):
+                        self.mcts.search(board, nnet)
 
+                    # Gets the moves and policy from the mcts, as well as the individual move to take
+                    # NOTE: moves[i] corresponds to search_probs[i]
+                    moves, search_probs, move = self.mcts.find_search_probs(fen_string, 5)
```

Comment:
If you don't create a variable for the parameter corresponding to `5`, please include a parameter label in the call to `find_search_probs` i.e.`find_search_probs(fen_string, temperature=5)`